Task3/MoneyManager/MoneyManager.DataAccess/Repositories/DataGenerator.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text;
+using MoneyManager;
+using System.IO;
+using System.Linq;
+using MoneyManager.DataAccess.Models;
+using Newtonsoft.Json;
+using Remotion.Linq.Clauses.ResultOperators;
+
+namespace MoneyManager.DataAccess.Repositories
+{
+    public class DataGenerator
+    {
+        private const string UserPath = "Users.txt";
+        private const string AssetPath = "Assets.txt";
+        private const string CategoryExpensesPath = "CategoryExpenses.json";
+        private const string CategoryIncomePath = "CategoryIncome.json";
+        private IApplicationContext ApplicationContext { get; }
+
+        public DataGenerator(IApplicationContext context)
+        {
+            ApplicationContext = context;
+        }
+
+        public void GenerateData()
+        {
+            var users = GenerateUsers();
+            var assets = GenerateAssets(ref users);
+            var categories = GenerateCategories();
+            var transactions = GenerateTransactions(ref categories, ref assets);
+            ApplicationContext.Users.AddRangeAsync(users);
+            ApplicationContext.Assets.AddRangeAsync(assets);
+            ApplicationContext.Categories.AddRangeAsync(categories);
+            ApplicationContext.Transactions.AddRange(transactions);
+            ApplicationContext.SaveChanges();
+        }
+
+        public List<User> GenerateUsers()
+        {
+            return File.ReadAllLines(UserPath).Select(strUser => strUser.Split(' '))
+                .Select(user => new User {Name = user[0], Email = user[1], Balance = 0, Assets = new List<Asset>()})
+                .ToList();
```

Comment:
I don't see any reason why these methods return `List` instead of `IEnumerable`.